app/components/provider_interface/change_offer_component.rb

```diff
@@ -29,6 +29,19 @@ def sub_component
         ProviderInterface::ChangeOffer::ChangeLocationComponent.new(
           change_offer_form: change_offer_form,
         )
+      when :study_mode
+        study_modes = Course.find(change_offer_form.course_id).study_modes_present
+        if study_modes.count > 1
+          ProviderInterface::ChangeOffer::ChangeStudyModeComponent.new(
+            change_offer_form: change_offer_form,
+          )
+        else # skip to location
```

Comment:
This works!

A small bikeshed: it's a bit of a shame to fix the order of steps out here. How do you feel about recursion? Not ideal with a method that relies on mutable object state like this one does, but it feels better not to duplicate the stepping logic

```ruby
change_offer_form.study_mode = study_modes.first
change_offer_form.step = change_offer_form.next_step
sub_component
```

Reply:
Hmm, recursion feels a bit riskier, but as long as `step` doesn't get stuck on `study_mode` we should be ok.